### PR TITLE
Task/enable visual snapshots cypress

### PR DESCRIPTION
**Pull request by shubhambajpaiNUK**
Description: I have taken the reference from Vinh's [PR:](https://github.com/newsuk/times-components/compare/task/TDP-778_enable_visual_snapshots?expand=1)  for enabling the cypress snapshot testing under SSR flow. 
I have worked over the flakiness of DropCap and ShareBar component in this PR. 
I am waiting for the contents to be fully loaded and disabling the cache in beforeEach segment and then added recursive function in order to retry the DropCap / Share Bar being loaded or not.

I have updated the snapshots and updated one command to updateScreenshot as a new job in package.json file.

### CODE REVIEW

File: packages/.env [hunk 1]
```diff
@@ -0,0 +1,10 @@
+STORYBOOK_ALGOLIA_ID=PZGYBTWG3J
+STORYBOOK_ALGOLIA_KEY=3835bd37d54757eda130c4055ca98c68
+STORYBOOK_ALGOLIA_INDEX=prod_articles_by_published_date_desc
+
+STORYBOOK_COMMENTING_CURRENT_ID=sp_rLv5PqMc
+STORYBOOK_COMMENTING_READONLY_ID=sp_pCQgrRiN
+STORYBOOK_COMMENTING_SWITCHOVER=2021-10-10T16:00:00.000Z
+
+GRAPHQL_ENDPOINT=https://api.thetimes.co.uk/graphql
+GRAPHQL_TOKEN=45aea5e6-e1be-11e7-80c1-9a214cf093ae
```
Comment (fayeez-tnl) [inline packages/.env:10]: Doesn't look like we should be checking these in?

File: packages/article-skeleton/src/article-skeleton.js [hunk 2]
```diff
@@ -11,7 +11,7 @@ import {
   AlgoliaSearchProvider
 } from "@times-components/ts-components";
 import UserState from "@times-components/user-state";
-import { MessageContext } from "@times-components/message-bar";
+import { MessageContext, MessageManager } from "@times-components/message-bar";
```
Comment (fayeez-tnl) [inline packages/article-skeleton/src/article-skeleton.js:14]: Why is this added here?

File: packages/article-skeleton/src/article-skeleton.js [hunk 3]
```diff
@@ -116,6 +116,7 @@ const ArticleSkeleton = ({
         }}
         analyticsStream={analyticsStream}
       >
+        <h1>{article.template}</h1>
```
Comment (fayeez-tnl) [inline packages/article-skeleton/src/article-skeleton.js:119]: How come you added this?

File: packages/ssr/__tests__/cypress/support/index.js [hunk 4]
```diff
@@ -19,6 +19,7 @@ import "cypress-axe";
 
 // Alternatively you can use CommonJS syntax:
 require("cypress-failed-log");
+require("cypress-wait-until");
```
Comment (fayeez-tnl) [inline packages/ssr/__tests__/cypress/support/index.js:22]: Why do we need this package? Can't we use cypress.wait command?

File: packages/ssr/__tests__/cypress/support/index.js [hunk 5]
```diff
@@ -33,3 +34,62 @@ Cypress.on("uncaught:exception", err => {
...
+      req.on('before:response', (res) => {
+        // force all API responses to not be cached
+        res.headers['cache-control'] = 'no-store'
+      })
+    }
+  ).as('interceptingImages')
+}
+
+export const checkDropCapChanges = (path) => {
+  cy.waitUntil(() => cy.get(path).should('be.visible') // yields <nav>
```
Comment (fayeez-tnl) [inline packages/ssr/__tests__/cypress/support/index.js:52]: can't we wait for the selector with cy.wait(..)?

File: packages/ssr/__tests__/cypress/support/index.js [hunk 6]
```diff
@@ -33,3 +34,62 @@ Cypress.on("uncaught:exception", err => {
...
+    .should('have.css', 'font-family') // yields 'sans-serif'
+    .and('match', /TimesModern-Regular/), {
+    errorMsg: "Element is not found even after waiting",
+    timeout: 10000,
+    interval: 500
+  })
+}
+
+export const checkShareBarLoaded = (path) => {
+  cy.waitUntil(() => cy.get(path).should('have.css', 'justify-content') // yields 'sans-serif'
```
Comment (fayeez-tnl) [inline packages/ssr/__tests__/cypress/support/index.js:62]: can't we wait for the selector with cy.wait(..)?

File: packages/ssr/__tests__/helpers/article-helper.js [hunk 7]
```diff
@@ -106,6 +116,49 @@ const articleTemplateTest = (template, options = {}) => {
       cy.get("script[data-spotim-module]").should("not.exist");
     });
 
+    it.only("should match snapshots", () => {
```
Comment (fayeez-tnl) [inline packages/ssr/__tests__/helpers/article-helper.js:119]: Remove the only here :)

File: packages/ssr/__tests__/cypress/support/index.js [hunk 8]
```diff
@@ -33,3 +34,62 @@ Cypress.on("uncaught:exception", err => {
...
+    // Requesting the page to reload (F5)
+    cy.reload(true);
+
+    // Wait a second for the server to respond and the DOM to be updated.
+    return cy.wait(1000).then(() => {
+      return waitUntilSelectorExists(skipDropCapCheck, remainingAttempts);
+    });
+  }
+  throw Error('Selector was not found.');
+}
```
Comment (fayeez-tnl) [inline packages/ssr/__tests__/cypress/support/index.js:95]: Let's grab some time to talk about this - I feel there may be a better way to do what you are trying to do.

**Review by fayeez-tnl — CHANGES_REQUESTED** [review]
Hey Shubham sorry for blocking the PR, let's talk through some of the changes here :)

File: packages/ssr/__tests__/cypress/support/index.js (hunk 8 above)
Comment (shubhambajpaiNUK) [reply packages/ssr/__tests__/cypress/support/index.js:95]: Sure Fayeez. Thanks for the review :)

File: packages/.env (hunk 1 above)
Comment (vinhlam) [reply packages/.env:10]: sorry, did i add that? it should be in the `.gitignore`

Comment (bajpaijee) [reply packages/.env:10]: Thanks @vinhlam, I will update and remove the file from the PR.

File: packages/ssr/__tests__/helpers/article-helper.js (hunk 7 above)
Comment (shubhambajpaiNUK) [reply packages/ssr/__tests__/helpers/article-helper.js]: It has been fixed

File: packages/article-skeleton/src/article-skeleton.js (hunk 3 above)
Comment (shubhambajpaiNUK) [reply packages/article-skeleton/src/article-skeleton.js]: I have removed the heading tag and updated the screenshots

File: packages/article-skeleton/src/article-skeleton.js (hunk 2 above)
Comment (shubhambajpaiNUK) [reply packages/article-skeleton/src/article-skeleton.js]: Removed

**Review by fayeez-tnl — APPROVED** [review]

**Pull request merged**